File: frontend/src/types/project.ts
export interface ProjectListItem {
  projectId: string
  code: string
  name: string
  address: string
  clientName?: string | null
  status: string
  createdAt: string
}

export interface ProjectDetail {
  projectId: string
  code: string
  name: string
  address: string
  description?: string | null
  clientName?: string | null
  status: string
  budget?: number | null
  startDate?: string | null
  endDate?: string | null
  createdAt: string
}

export interface CreateProjectPayload {
  code: string
  name: string
  address: string
  description?: string | null
  clientName?: string | null
  status: string
  budget?: number | null
  startDate?: string | null
  endDate?: string | null
}

export interface UpdateProjectPayload {
  code: string
  name: string
  address: string
  description?: string | null
  clientName?: string | null
  status: string
  budget?: number | null
  startDate?: string | null
  endDate?: string | null
}

export interface AiWeeklySummaryRequest {
  contextNotes?: string | null
}

export interface AiWeeklySummary {
  headline: string
  summary: string
  progressHighlights: string[]
  riskAlerts: string[]
  nextWeekPlan: string[]
  openDecisions: string[]
}

File: backend/ConstructionManagement/Entities/Project.cs
namespace ConstructionManagement.Entities
{
    public class Project
    {
        public Guid ProjectId { get; set; }

        public required string Code { get; set; }

        public required string Name { get; set; }

        public required string Address { get; set; }

        public string? Description { get; set; }

        public string? ClientName { get; set; }

        public string Status { get; set; } = "Planning";

        public decimal? Budget { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<TaskItem> TaskItems { get; set; } = new List<TaskItem>();

        public ICollection<Variation> Variations { get; set; } = new List<Variation>();

        public ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
    }
}
